File: introduction/ITP1_10_C-Standard Deviation.js
var lines = require('fs').readFileSync('/dev/stdin', 'utf8').trim().split('\n');
while(lines[0] != '0') {
  var n = lines.shift(), m = 0, sum = 0;
  var arr = lines.shift().split(' ').map(Number);
  for(var i = 0; i < n; i++) {
    s = arr[i];
    m += s;
    sum += s*s;
  }
  m /= n;
  console.log(Math.sqrt(sum/n-m*m));
}

File: introduction/ITP1_10_C-Standard Deviation.scala
object Main extends App {
  val lines = io.Source.stdin.getLines
  var n = lines.next.toInt
  while(n != 0) {
    val s = lines.next.split(" ").map(_.toDouble)
    val m = s.sum / n
    println(Math.sqrt(s.foldLeft(0d)((x,y)=>x+(y-m)*(y-m))/n))
    n = lines.next.toInt
  }
}
